### PR TITLE
Run pack acceptance tests on pushes to main & release/*

### PR DESCRIPTION
**Pull request by natalieparellano**
Signed-off-by: Natalie Arellano <narellano@vmware.com>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -35,7 +35,6 @@ jobs:
           file: ./out/tests/coverage-unit.txt
           flags: unit,os_linux
           fail_ci_if_error: true
-          verbose: true
```
Comment (natalieparellano) [inline .github/workflows/build.yml:38]: Unrelated change, to suppress a warning.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -113,11 +126,110 @@ jobs:
...
+  pack-acceptance-linux:
+    if: github.event_name == 'push'
+    needs: build-and-publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'buildpacks/pack'
+          path: 'pack'
+          ref: 'main'
```
Comment (jabrown85) [inline .github/workflows/build.yml:151]: Could this reference the latest release? Or do you explicitly want `main`?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -113,11 +126,110 @@ jobs:
...
+          cat tag.txt >> $GITHUB_ENV
+      - uses: actions/download-artifact@v2
+        with:
+          name: lifecycle-linux-x86-64
+          path: pack
+      - name: Run pack acceptance
+        run: |
+          cd pack
+          git checkout $(git describe --abbrev=0 --tags) # check out the latest tag
+          LIFECYCLE_PATH="../lifecycle-v${{ env.LIFECYCLE_VERSION }}+linux.x86-64.tgz" \
```
Comment (jabrown85) [inline .github/workflows/build.yml:171]: Why does pack need two different references to the same lifecycle?

File: .github/workflows/post-release.yml [hunk 4]
```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Set env
         run: |
           echo "LIFECYCLE_VERSION=$(echo ${{ github.event.release.tag_name }} | cut -d "v" -f2)" >> $GITHUB_ENV
-          echo "LIFECYCLE_IMAGE_TAG=$(git describe --always --dirty)" >> $GITHUB_ENV
+          echo "LIFECYCLE_IMAGE_TAG=$(git describe --always)" >> $GITHUB_ENV
```
Comment (jabrown85) [inline .github/workflows/post-release.yml:20]: Removing `--dirty` makes sense - was it there to guard producing images for a repo with uncommitted code?

**Review by jabrown85 — APPROVED** [review]

File: .github/workflows/post-release.yml (hunk 4 above)
Comment (natalieparellano) [reply .github/workflows/post-release.yml:20]: Way way back when we first decided to auto build & tag lifecycle images, there was the concept of `--dirty` because we thought that would be useful in local development. But it turned out to be more confusing than useful, and we dropped it. I don't think we ever want to publish `<some-sha>-dirty` tagged images to buildpacksio, though it could happen go.sum changes are uncommitted, hence this removal.

File: .github/workflows/build.yml (hunk 3 above)
Comment (natalieparellano) [reply .github/workflows/build.yml:171]: `LIFECYCLE_PATH` is used to create the builder that `pack` uses for trusted builder workflows. `LIFECYCLE_IMAGE` is for the untrusted workflows. I don't think pack lets you grab the lifecycle for a builder from a lifecycle image (though that'd be nice).

File: .github/workflows/build.yml (hunk 2 above)
Comment (natalieparellano) [reply .github/workflows/build.yml:151]: I think this is made superfluous by `git checkout $(git describe --abbrev=0 --tags)`.
